src/context/ApiContext.tsx

```diff
@@ -126,42 +129,45 @@ export const ApiContext = React.createContext<IApiContext>({
-  putUserSubjects: reject,
-  putUserProjectFields: reject,
-  becomeInstructor: reject,
-  putUserActiveFalse: reject,
-  createCertificate: reject,
-  getCertificate: reject,
-  getCertificates: reject,
-  getCourses: reject,
-  getCourseTags: reject,
-  getCourse: reject,
-  getMyCourses: reject,
-  createCourse: reject,
-  cancelCourse: reject,
-  editCourse: reject,
-  joinCourse: reject,
-  leaveCourse: reject,
-  submitCourse: reject,
-  publishSubCourse: reject,
-  createSubCourse: reject,
-  editSubCourse: reject,
-  cancelSubCourse: reject,
-  createLecture: reject,
-  editLecture: reject,
-  cancelLecture: reject,
-  registerTutee: reject,
-  registerStateTutee: reject,
-  registerTutor: reject,
-  sendCourseGroupMail: reject,
-  joinBBBmeeting: reject,
-  getCooperatingSchools: reject,
-  getMentoringMaterial: reject,
-  getFeedbackCallData: reject,
-  postContactMentor: reject,
-  postUserRoleProjectCoach: reject,
-  postUserRoleProjectCoachee: reject,
+  putUser: (user) => Promise.reject(),
+  putUserSubjects: (subjects) => Promise.reject(),
+  putUserProjectFields: (projectFields) => Promise.reject(),
+  becomeInstructor: (description: BecomeInstructor | BecomeIntern) =>
+    Promise.reject(),
+  putUserActiveFalse: () => Promise.reject(),
+  getCertificate: (cerfiticateData) => Promise.reject(),
+  getCourses: () => Promise.reject(),
+  getCourseTags: () => Promise.reject(),
+  getCourse: (courseId) => Promise.reject(),
+  getMyCourses: () => Promise.reject(),
+  createCourse: (course) => Promise.reject(),
+  cancelCourse: (id) => Promise.reject(),
+  editCourse: (id, course) => Promise.reject(),
+  joinCourse: (courseId: number, subCourseId: number, participantId: string) =>
+    Promise.reject(),
+  leaveCourse: (courseId: number, subCourseId: number, participantId: string) =>
+    Promise.reject(),
+  submitCourse: (id, course) => Promise.reject(),
+  publishSubCourse: (courseId, id, course) => Promise.reject(),
+  createSubCourse: (id, subCourse) => Promise.reject(),
+  editSubCourse: (id, subCourse) => Promise.reject(),
+  cancelSubCourse: (id, subCourseId) => Promise.reject(),
+  createLecture: (id, subCourseId, lecture) => Promise.reject(),
+  editLecture: (id, subCourseId, lecture) => Promise.reject(),
+  cancelLecture: (id, subCourseId, lectureId) => Promise.reject(),
+  registerTutee: (tutee) => Promise.reject(),
+  registerStateTutee: (tutee) => Promise.reject(),
+  registerTutor: (tutor) => Promise.reject(),
+  sendCourseGroupMail: (id, subCourseId, subject, body) => Promise.reject(),
+  joinBBBmeeting: (courseId, subcourseId) => Promise.reject(),
+  getCooperatingSchools: (state) => Promise.reject(),
+  getMentoringMaterial: (type, location) => Promise.reject(),
+  getFeedbackCallData: () => Promise.reject(),
+  postContactMentor: (message) => Promise.reject(),
+  postUserRoleProjectCoach: () => Promise.reject(),
+  postUserRoleProjectCoachee: () => Promise.reject(),
+  postRequestCode: () => Promise.reject(),
```

Comment:
In the definition of `IApiContext` `postRequestCode` and `postVerifyCode` got arguments which isn't the case here. That leads to the following build error on heroku:

> -----> React.js (create-react-app) multi app detected
> =====> Downloading Buildpack: https://github.com/heroku/heroku-buildpack-multi.git
> =====> Detected Framework: Multipack
> =====> Downloading Buildpack: https://github.com/heroku/heroku-buildpack-nodejs.git
> =====> Detected Framework: Node.js
>        
> -----> Creating runtime environment
>        
>        NPM_CONFIG_LOGLEVEL=error
>        USE_YARN_CACHE=true
>        NODE_ENV=production
>        NODE_MODULES_CACHE=true
>        NODE_VERBOSE=false
>        
> -----> Installing binaries
>        engines.node (package.json):  unspecified
>        engines.npm (package.json):   unspecified (use default)
>        engines.yarn (package.json):  unspecified (use default)
>        
>        Resolving node version 12.x...
>        Downloading and installing node 12.20.0...
>        Using default npm version: 6.14.8
>        Resolving yarn version 1.22.x...
>        Downloading and installing yarn (1.22.10)
>        Installed yarn 1.22.10
>        
> -----> Restoring cache
>        - yarn cache
>        
> -----> Installing dependencies
>        Installing node modules (yarn.lock)
>        yarn install v1.22.10
>        [1/4] Resolving packages...
>        [2/4] Fetching packages...
>        info fsevents@2.1.2: The platform "linux" is incompatible with this module.
>        info "fsevents@2.1.2" is an optional dependency and failed compatibility check. Excluding it from installation.
>        info fsevents@1.2.13: The platform "linux" is incompatible with this module.
>        info "fsevents@1.2.13" is an optional dependency and failed compatibility check. Excluding it from installation.
>        info fsevents@2.1.3: The platform "linux" is incompatible with this module.
>        info "fsevents@2.1.3" is an optional dependency and failed compatibility check. Excluding it from installation.
>        [3/4] Linking dependencies...
>        warning " > @testing-library/user-event@7.2.1" has unmet peer dependency "@testing-library/dom@>=5".
>        warning " > eslint-config-react-app@5.2.1" has incorrect peer dependency "@typescript-eslint/eslint-plugin@2.x".
>        warning " > eslint-config-react-app@5.2.1" has incorrect peer dependency "@typescript-eslint/parser@2.x".
>        warning " > eslint-config-react-app@5.2.1" has unmet peer dependency "babel-eslint@10.x".
>        warning " > eslint-config-react-app@5.2.1" has incorrect peer dependency "eslint-plugin-flowtype@3.x || 4.x".
>        warning " > eslint-config-react-app@5.2.1" has incorrect peer dependency "eslint-plugin-react-hooks@1.x || 2.x".
>        warning " > styled-components@5.1.1" has unmet peer dependency "react-is@>= 16.8.0".
>        warning " > eslint-config-airbnb@18.2.0" has incorrect peer dependency "eslint@^5.16.0 || ^6.8.0 || ^7.2.0".
>        warning "eslint-config-airbnb > eslint-config-airbnb-base@14.2.0" has incorrect peer dependency "eslint@^5.16.0 || ^6.8.0 || ^7.2.0".
>        warning " > eslint-plugin-flowtype@5.2.0" has incorrect peer dependency "eslint@^7.1.0".
>        [4/4] Building fresh packages...
>        Done in 37.36s.
>        
> -----> Build
>        Running build (yarn)
>        yarn run v1.22.10
>        $ react-scripts build
>        Creating an optimized production build...
>        Failed to compile.
>        
>        /tmp/build_9c31f701_/src/context/ApiContext.tsx
>        TypeScript error in /tmp/build_9c31f701_/src/context/ApiContext.tsx(127,60):
>        Argument of type '{ getUserData: () => Promise<never>; dissolveMatch: () => Promise<never>; dissolveProjectMatch: () => Promise<never>; requestNewToken: (email: string, redirectTo: string) => Promise<...>; ... 35 more ...; postVerifyCode: () => Promise<...>; }' is not assignable to parameter of type 'IApiContext'.
>          Type '{ getUserData: () => Promise<never>; dissolveMatch: () => Promise<never>; dissolveProjectMatch: () => Promise<never>; requestNewToken: (email: string, redirectTo: string) => Promise<...>; ... 35 more ...; postVerifyCode: () => Promise<...>; }' is missing the following properties from type 'IApiContext': createCertificate, getCertificates  TS2345
>        
>            125 | const reject = () => Promise.reject();
>            126 | 
>          > 127 | export const ApiContext = React.createContext<IApiContext>({
>                |                                                            ^
>            128 |   getUserData: reject,
>            129 |   dissolveMatch: reject,
>            130 |   dissolveProjectMatch: reject,
>        
>        
> error Command failed with exit code 1.
>        info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
> -----> Build failed
>        
>        We're sorry this build is failing! You can troubleshoot common issues here:
>        https://devcenter.heroku.com/articles/troubleshooting-node-deploys
>        
>        Some possible problems:
>        
>        - Node version not specified in package.json
>          https://devcenter.heroku.com/articles/nodejs-support#specifying-a-node-js-version
>        
>        Love,
>        Heroku
>        
>  !     Push rejected, failed to compile React.js (create-react-app) multi app.
>  !     Push failed

Reply:
@lsc0498 not quite, a function type with less parameters is assignable to a function type with more parameters (the inverse is not true, except the parameters are optional). The actual cause here was some APIs added which were lost during the merge, but that was already corrected by Marcel :) (I admit that the Typescript error messages are sometimes quite hard to read ...)